peachjam/js/components/DocumentSearch/HTMLResults.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <div
+    v-for="(snippet, index) in snippets"
+    :key="index"
+    class="card mb-2"
+  >
+    <div class="card-body">
+      <ResultSnippet
+        class="mb-2"
+        :node="snippet.cloneNode(true)"
+      />
+      <div>
+        <a
+          href="#"
+          @click.prevent="$emit('go-to-result', snippet);"
+        >
+          Go to result
+        </a>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import ResultSnippet from './ResultSnippet.vue';
+export default {
+  name: 'HTMLResults',
+  components: {
+    ResultSnippet
+  },
+  props: {
+    results: {
+      type: Array,
+      required: true
+    },
+    q: {
+      type: String,
+      required: false,
+      default: ''
+    }
+  },
+  emits: ['go-to-result'],
+  data: () => ({
+    snippets: []
+  }),
+  watch: {
+    q: {
+      handler () {
+        this.renderSnippets();
+      }
+    }
+  },
+
+  mounted () {
+    this.renderSnippets();
+  },
+  methods: {
+    renderSnippets () {
+      this.snippets = this.results.map(node => {
```

Comment:
De-duping the snippets (but preserving document order) would prevent duplicate snippets where there are multiple marks within the same parent snippet.

Reply:
done